### ISSUE TITLE
Cannot change request headers using cy.visit() options

### ISSUE THREAD

**Issue by warpdesign** [cypress-io/cypress#9641]
### Current behavior

Calling `cy.visit` with headers in options won't send these headers.

### Desired behavior

Adding headers to the visit `options` parameter is [supposed](https://docs.cypress.io/api/commands/visit.html) to add request headers:
```js
cy.visit('https://cypress.io',{
  'User-Agent': 'Mozilla/5.0 (iPhone; CPU iPhone OS 13_2_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.3 Mobile/15E148 Safari/604.1',
  'My-Header': 'foo'
})
```

### Test code to reproduce

Test repo: https://github.com/warpdesign/cypress-change-user-agent

When inspecting the request for `cypress.io` in the console, the `User Agent` hasn't been modified, and the `My-Header` does not appear:

![image](https://user-images.githubusercontent.com/199648/101894766-ec4cd400-3ba6-11eb-8a36-0514618e7b2a.png)

### Versions

Cypress 6.1.0
macOS 10.15.6
Chrome  87.0.4280.88

**Comment by jennifer-shehane** [comment]
If you're trying to set the user-agent, this needs to be done through Cypress config, since this requires being passed in when the browser initially launches. https://on.cypress.io/visit#User-agent

**Comment by warpdesign** [comment]
I think this was closed too fast: in my example I also set a new `My-Header` header which does not appear in the request header. This has nothing to do with the user agent.

**Comment by warpdesign** [comment]
Btw, according to change log of version [3.3.3](user-agent headers sent as arguments to cy.request() and cy.visit() are now properly sent. Fixes #3873 and #3719.) `user-agent` headers sent to cy.visit should be sent.

Looks like it's a regression to me.

**Comment by jennifer-shehane** [comment]
Yeah, it does seem like we expanded this. Where are you reading the values of the headers after, where they're not affected? I see that we have tests for setting headers on cy.visit() here: https://github.com/cypress-io/cypress/blob/develop/packages/driver/cypress/integration/commands/navigation_spec.js#L783:L783 where the value of the headers is printed correctly https://github.com/cypress-io/cypress/blob/develop/packages/driver/cypress/plugins/server.js#L133:L133
